### PR TITLE
Refreshing Tags Editor: Mark 3

### LINKED ISSUE #57: Make it possible to expand / increase tags area height
<a href="https://github.com/bryceadams"></img></a> **Issue by [bryceadams](https://github.com/bryceadams)**
_Wednesday Oct 07, 2015 at 22:24 GMT_

---

It'd be great if it were possible to expand the height of the tags area so it would be possible to view multiple lines of tags at once. I imagine this isn't much of an issue with normal textual tags but if you're collaborating with 5+ people, the emails push it over the first line quite quickly.

screen

### PR DESCRIPTION
**Pull request by jleandroperez**
### Description:
In this PR we're turning the Tags Editor into a scroll friendly component.

@emilylaguna May I pick your brain with this one?
Would love to hear what you think about... oh well, the whole mechanism!!

**Thank you Em!!!**

Closes #57
Ref. #458 

### Details:
- This PR wraps **TagsField** into a **NSScrollView**, and implements intrinsicContentSize
- All of the workarounds we're using are properly documented in the code itself
- We're also implementing a mechanism to prevent the cursor from falling offscreen

### Test
- [ ] Verify the tags editor displays a placeholder when there are no tags: `Add tag...`
- [ ] Verify that whenever you type a word (on an empty Tags Field), the baseline is preserved
- [ ] Verify that whenever you commit your first tag (press **return**!) the baseline is also preserved
- [ ] Verify that whenever you add enough tags to fill the editor, scrolling is allowed!
- [ ] Verify that moving around with the keyboard's arrows causes the field to autoscroll
- [ ] Verify that double clicking to edit the (single) Tag present in a note, switches to Edition mode, and the text is vertically centered

### Release
`RELEASE-NOTES.txt` was updated in 2c64cec with:
 
> New Scroll Support in the Tags Editor area

### CODE REVIEW

File: Simplenote/TagsField.swift [hunk 1]
```diff
@@ -170,6 +221,39 @@ extension TagsField: NSTextViewDelegate {
...
+        /// Determine the Editor's cursor location
+        ///
+        var output = lm.boundingRect(forGlyphRange: textView.selectedRange(), in: container)
+
+        let placeholderWidth = simplenotePlaceholderAttributedString.size().width
+
+        /// Adjust the Viewport: always accommodate to support `placeholder.width`'s on both sides
+        ///
+        output.origin.x = max(output.origin.x - placeholderWidth, .zero)
+        output.size.width = simplenotePlaceholderAttributedString.size().width * 2
```
[the remark this answers is not in the document]
Comment (jleandroperez) [reply Simplenote/TagsField.swift:250]: Ouch!!! totally missed that line, thank you!!

**Comment by jleandroperez** [comment]
@emilylaguna ready for another look, **Thank YOU!!!**

File: Simplenote/SPHorizontalScrollView.h [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+// MARK: - SPHorizontalScrollView
+//          This NSScrollView subclass remaps Vertical Scroll events into Horizontal Scroll events, in order to
+//          support ScrollWheel events performed with a mouse (single axis device!).
+//
+@interface SPHorizontalScrollView : NSScrollView
```
Comment (emilylaguna) [inline Simplenote/SPHorizontalScrollView.h:9]: Any reason why you chose to write this in Objective-C?

**Comment by jleandroperez** [comment]
@emilylaguna **Thank you** for swiftifying HorizontalScrollView!!!

All changes UP, ready for another peek. 🔥

**Comment by jleandroperez** [comment]
Wooooo thank you Emily!!!

**Pull request merged**